pkg/clusterapi/aws/actuator_test.go

```diff
@@ -0,0 +1,338 @@
+				} else {
+					assertRunInstancesInputHasTag(t, runInput, "sub-host-type", "compute")
+				}
+
+				assertRunInstancesInputHasTag(t, runInput, "kubernetes.io/cluster/"+cluster.Name, cluster.Name)
+				assertRunInstancesInputHasTag(t, runInput, "Name", machine.Name)
+
+				assert.Equal(t, testImage, *runInput.ImageId)
+			}).Return(&ec2.Reservation{
+				Instances: []*ec2.Instance{{InstanceId: aws.String("newinstance")}},
+			}, nil)
+
+			actuator := NewActuator(kubeClient, capiClient, log.WithField("test", "TestActuator"), "us-east-1c")
+			actuator.clientBuilder = func(kubeClient kubernetes.Interface, mSpec *clustopv1.MachineSetSpec, namespace, region string) (AWSClient, error) {
+				return mockAWSClient, nil
+			}
+
+			instance, err := actuator.CreateMachine(cluster, machine)
+			assert.NoError(t, err)
+			assert.NotNil(t, instance)
+		})
+	}
+}
+
+func assertRunInstancesInputHasTag(t *testing.T, input *ec2.RunInstancesInput, key, value string) {
+	for _, tag := range input.TagSpecifications[0].Tags {
+		if *tag.Key == key {
+			assert.Equal(t, *tag.Value, value, "bad value for instance tag: %s", key)
+			return
+		}
+	}
+	t.Errorf("tag not found on RunInstancesInput: %s", key)
+}
+
+func addDescribeImagesMock(mockAWSClient *MockAWSClient, imageID string) {
+	mockAWSClient.EXPECT().DescribeImages(&ec2.DescribeImagesInput{
+		ImageIds: []*string{aws.String(testImage)},
+	}).Return(
+		&ec2.DescribeImagesOutput{
+			Images: []*ec2.Image{
+				{
+					ImageId: aws.String(testImage),
+				},
+			},
+		}, nil)
+}
+
+func addDescribeVpcsMock(mockAWSClient *MockAWSClient, vpcName, vpcID string) {
+	describeVpcsInput := ec2.DescribeVpcsInput{
+		Filters: []*ec2.Filter{{Name: aws.String("tag:Name"), Values: []*string{&vpcName}}},
+	}
+	describeVpcsOutput := ec2.DescribeVpcsOutput{
+		Vpcs: []*ec2.Vpc{
+			{
+				VpcId: aws.String(vpcID),
+			},
+		},
+	}
+	mockAWSClient.EXPECT().DescribeVpcs(&describeVpcsInput).Return(&describeVpcsOutput, nil)
+}
+
+func addDescribeSubnetsMock(mockAWSClient *MockAWSClient, az, vpcID, subnetID string) {
+	input := ec2.DescribeSubnetsInput{
+		Filters: []*ec2.Filter{
+			{Name: aws.String("vpc-id"), Values: []*string{aws.String(vpcID)}},
+			{Name: aws.String("availability-zone"), Values: []*string{aws.String(az)}},
+		},
+	}
+	output := ec2.DescribeSubnetsOutput{
+		Subnets: []*ec2.Subnet{
+			{
+				SubnetId: aws.String(subnetID),
+			},
+		},
+	}
+	mockAWSClient.EXPECT().DescribeSubnets(&input).Return(&output, nil)
+}
+
+func addDescribeSecurityGroupsMock(mockAWSClient *MockAWSClient, vpcID, vpcName string, isMaster, isInfra bool) {
+	// Using the input builder from the production code. Behavior of this function is tested separately.
+	input := buildDescribeSecurityGroupsInput(vpcID, vpcName, isMaster, isInfra)
+	output := ec2.DescribeSecurityGroupsOutput{
+		SecurityGroups: []*ec2.SecurityGroup{},
+	}
+	// Using the explicit index where we store our name filters, may want to search for it if this
```

Comment:
As noted in previous comments from earlier in the file, I think we should search explicitly for the group-name filters. I don't think that the amount of effort required justifies the fragility,